com.unity.render-pipelines.universal/Runtime/UniversalGlobalSettings.cs

```diff
@@ -0,0 +1,179 @@
+                }
+            }
+
+            if (assetCreated)
+            {
+                if (src != null)
+                {
+                    assetCreated.lightLayerName0 = System.String.Copy(src.lightLayerName0);
+                    assetCreated.lightLayerName1 = System.String.Copy(src.lightLayerName1);
+                    assetCreated.lightLayerName2 = System.String.Copy(src.lightLayerName2);
+                    assetCreated.lightLayerName3 = System.String.Copy(src.lightLayerName3);
+                    assetCreated.lightLayerName4 = System.String.Copy(src.lightLayerName4);
+                    assetCreated.lightLayerName5 = System.String.Copy(src.lightLayerName5);
+                    assetCreated.lightLayerName6 = System.String.Copy(src.lightLayerName6);
+                    assetCreated.lightLayerName7 = System.String.Copy(src.lightLayerName7);
+                }
+                else
+                {
+                }
+
+                AssetDatabase.SaveAssets();
+                AssetDatabase.Refresh();
+            }
+
+            return assetCreated;
+        }
+
+#endif
+
+        void Init()
+        {
+            ResetLightLayerNames();
+        }
+
+        #region Light Layer Names [Light]
+
+        /// <summary>Name for light layer 0.</summary>
+        public string lightLayerName0;
+        /// <summary>Name for light layer 1.</summary>
+        public string lightLayerName1;
+        /// <summary>Name for light layer 2.</summary>
+        public string lightLayerName2;
+        /// <summary>Name for light layer 3.</summary>
+        public string lightLayerName3;
+        /// <summary>Name for light layer 4.</summary>
+        public string lightLayerName4;
+        /// <summary>Name for light layer 5.</summary>
+        public string lightLayerName5;
+        /// <summary>Name for light layer 6.</summary>
+        public string lightLayerName6;
+        /// <summary>Name for light layer 7.</summary>
+        public string lightLayerName7;
+
+
+        [System.NonSerialized]
+        string[] m_LightLayerNames = null;
+        /// <summary>
+        /// Names used for display of light layers.
+        /// </summary>
+        public string[] lightLayerNames
+        {
+            get
+            {
+                if (m_LightLayerNames == null)
+                {
+                    m_LightLayerNames = new string[8];
+                }
+
+                m_LightLayerNames[0] = lightLayerName0;
+                m_LightLayerNames[1] = lightLayerName1;
+                m_LightLayerNames[2] = lightLayerName2;
+                m_LightLayerNames[3] = lightLayerName3;
+                m_LightLayerNames[4] = lightLayerName4;
+                m_LightLayerNames[5] = lightLayerName5;
+                m_LightLayerNames[6] = lightLayerName6;
+                m_LightLayerNames[7] = lightLayerName7;
+
+                return m_LightLayerNames;
+            }
```

Comment:
It's not ideal for a getter to mutate state of an object. A better solution imo would be to serialize the light name array and change the editor to display the array property nice.

Reply:
the reason it was done this way is also for when people want to change the layer names by script. they can do it by calling "lightLayerName3" for example but internally we serialize an array. i am open to suggestion though, this was implemented that way in HDRP.